src/components/PopupWithForm.js

```diff
@@ -8,6 +8,17 @@ export default class PopupWithForm extends Popup {
       ...this._popupForm.querySelectorAll(".modal__form-input"),
     ];
     this._handleFormSubmit = handleFormSubmit;
+
+    this._popupButton = this._popupForm.querySelector(".modal__form-button");
+    this._submitButtonText = this._popupButton.textContent;
+  }
+
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // Here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`